### PR TITLE
app-emulation/spice-vdagent: Bump to version 0.22.1

### PR DESCRIPTION
**Pull request by Thamognya**
- bump to version 0.22.1
- eapi 8

Signed-off-by: Thamognya Kodi <contact@thamognya.com>

### CODE REVIEW

**Comment by thesamesam** [comment]
Could you squash both commits into the first one?

**Comment by ThamognyaKodi** [comment]
yup, will work on that.

**Comment by thesamesam** [comment]
Let me know if need help/more detail!

File: app-emulation/spice-vdagent/spice-vdagent-0.22.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+		--with-init-script=systemd \
+		--localstatedir="${EPREFIX}"/var \
+		$(use_with gtk) \
+		"${opt[@]}"
+}
+
+src_install() {
+	default
+
+	cd "${ED}" && rmdir -p var/run/spice-vdagentd || die
```
Comment (juippis) [inline app-emulation/spice-vdagent/spice-vdagent-0.22.1.ebuild:57]:
```suggestion
	rmdir -p "${ED}" /var/run/spice-vdagentd || die
```

File: app-emulation/spice-vdagent/spice-vdagent-0.22.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	x11-libs/libXfixes
+	x11-libs/libXrandr
+	x11-libs/libX11
+	x11-libs/libXinerama
+	gtk? ( x11-libs/gtk+:3 )
+	systemd? ( sys-apps/systemd )"
+DEPEND="${CDEPEND}
+	virtual/pkgconfig"
+RDEPEND="${CDEPEND}
+	selinux? ( sec-policy/selinux-vdagent )"
```
Comment (juippis) [inline app-emulation/spice-vdagent/spice-vdagent-0.22.1.ebuild:33]:
```suggestion
DEPEND="
	dev-libs/glib:2
	>=app-emulation/spice-protocol-0.14.0
	media-libs/alsa-lib
	sys-apps/dbus
	x11-libs/libdrm
	>=x11-libs/libpciaccess-0.10
	x11-libs/libXfixes
	x11-libs/libXrandr
	x11-libs/libX11
	x11-libs/libXinerama
	gtk? ( x11-libs/gtk+:3 )
	systemd? ( sys-apps/systemd )"
RDEPEND="${DEPEND}
	selinux? ( sec-policy/selinux-vdagent )"
BDEPEND="virtual/pkgconfig"
```

`pkgconfig` is a BDEPEND and therefore we can delete the "COMMON_DEPEND" condition here.

**Review by juippis — COMMENTED** [review]
Now one thing I don't really get:
```
 * QA Notice: Unrecognized configure options:
 * 
 * 	--disable-static
 * 	--disable-static
```
I wonder if tinderbox will report this? @gentoo/qa do you know?

File: app-emulation/spice-vdagent/spice-vdagent-0.22.1.ebuild (hunk 2 above)
Comment (ThamognyaKodi) [reply app-emulation/spice-vdagent/spice-vdagent-0.22.1.ebuild]: I see, I think I have made the appropriate changes recommended now.

File: app-emulation/spice-vdagent/spice-vdagent-0.22.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+		--with-init-script=systemd \
+		--localstatedir="${EPREFIX}"/var \
+		$(use_with gtk) \
+		"${opt[@]}"
+}
+
+src_install() {
+	default
+
+	rmdir -p "${ED}" /var/run/spice-vdagentd || die
```
Comment (juippis) [inline app-emulation/spice-vdagent/spice-vdagent-0.22.1.ebuild:56]:
```suggestion
	rmdir -p "${ED}"/var/run/spice-vdagentd || die
```

Although:
`rmdir: failed to remove directory '/var/tmp/portage/app-emulation/spice-vdagent-0.22.1/image': Directory not empty`
Is this supposed to be removed? Can you add a comment above why.

File: app-emulation/spice-vdagent/spice-vdagent-0.22.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+		--with-init-script=systemd \
+		--localstatedir="${EPREFIX}"/var \
+		$(use_with gtk) \
+		"${opt[@]}"
+}
+
+src_install() {
+	default
+
+	rmdir -p "${ED}"/var/run/spice-vdagentd || die
```
Comment (juippis) [inline app-emulation/spice-vdagent/spice-vdagent-0.22.1.ebuild:56]: Again, this fails.
```
>>> Install app-emulation/spice-vdagent-0.22.1 into /var/tmp/portage/app-emulation/spice-vdagent-0.22.1/image
make -j16 DESTDIR=/var/tmp/portage/app-emulation/spice-vdagent-0.22.1/image install 
make[1]: Entering directory '/var/tmp/portage/app-emulation/spice-vdagent-0.22.1/work/spice-vdagent-0.22.1'
/bin/mkdir -p /var/tmp/portage/app-emulation/spice-vdagent-0.22.1/image/var/run/spice-vdagentd
 /bin/mkdir -p '/var/tmp/portage/app-emulation/spice-vdagent-0.22.1/image/usr/bin'
 /bin/mkdir -p '/var/tmp/portage/app-emulation/spice-vdagent-0.22.1/image/usr/sbin'
 /bin/mkdir -p '/var/tmp/portage/app-emulation/spice-vdagent-0.22.1/image/usr/share/gdm/greeter/autostart'
 /bin/mkdir -p '/var/tmp/portage/app-emulation/spice-vdagent-0.22.1/image/usr/share/gdm/autostart/LoginWindow'
 /bin/mkdir -p '/var/tmp/portage/app-emulation/spice-vdagent-0.22.1/image/usr/share/man/man1'
 /bin/mkdir -p '/var/tmp/portage/app-emulation/spice-vdagent-0.22.1/image/usr/lib/tmpfiles.d'
 /usr/lib/portage/python3.10/ebuild-helpers/xattr/install -c -m 644 ./data/spice-vdagent.desktop '/var/tmp/portage/app-emulation/spice-vdagent-0.22.1/image/usr/share/gdm/greeter/autostart'
 /usr/lib/portage/python3.10/ebuild-helpers/xattr/install -c -m 644 ./data/spice-vdagent.desktop '/var/tmp/portage/app-emulation/spice-vdagent-0.22.1/image/usr/share/gdm/autostart/LoginWindow'
 /bin/mkdir -p '/var/tmp/portage/app-emulation/spice-vdagent-0.22.1/image/lib/udev/rules.d'
 /usr/lib/portage/python3.10/ebuild-helpers/xattr/install -c -m 644 ./data/tmpfiles.d/spice-vdagentd.conf '/var/tmp/portage/app-emulation/spice-vdagent-0.22.1/image/usr/lib/tmpfiles.d'
 /bin/mkdir -p '/var/tmp/portage/app-emulation/spice-vdagent-0.22.1/image/etc/xdg/autostart'
  /usr/lib/portage/python3.10/ebuild-helpers/xattr/install -c src/spice-vdagent '/var/tmp/portage/app-emulation/spice-vdagent-0.22.1/image/usr/bin'
  /usr/lib/portage/python3.10/ebuild-helpers/xattr/install -c src/spice-vdagentd '/var/tmp/portage/app-emulation/spice-vdagent-0.22.1/image/usr/sbin'
 /usr/lib/portage/python3.10/ebuild-helpers/xattr/install -c -m 644 data/spice-vdagent.1 data/spice-vdagentd.1 '/var/tmp/portage/app-emulation/spice-vdagent-0.22.1/image/usr/share/man/man1'
 /usr/lib/portage/python3.10/ebuild-helpers/xattr/install -c -m 644 ./data/70-spice-vdagentd.rules '/var/tmp/portage/app-emulation/spice-vdagent-0.22.1/image/lib/udev/rules.d'
 /usr/lib/portage/python3.10/ebuild-helpers/xattr/install -c -m 644 ./data/spice-vdagent.desktop '/var/tmp/portage/app-emulation/spice-vdagent-0.22.1/image/etc/xdg/autostart'
make[1]: Leaving directory '/var/tmp/portage/app-emulation/spice-vdagent-0.22.1/work/spice-vdagent-0.22.1'
rmdir: failed to remove directory '/var/tmp/portage/app-emulation/spice-vdagent-0.22.1/image': Directory not empty
 * ERROR: app-emulation/spice-vdagent-0.22.1::gentoo failed (install phase):
 *   (no error message)
 * 
```
Does it work for you? 

```
       -p, --parents
              remove  DIRECTORY and its ancestors; e.g., 'rmdir -p a/b/c' is similar to 'rmdir a/b/c
              a/b a'
```
`-p` is not the flag you want to be using here.

Please do test your ebuild before pushing to make sure it works as you intend it to :)

**Comment by thesamesam** [comment]
ping @ThamognyaKodi

**Review by juippis — APPROVED** [review]
```
 * QA Notice: package is installing udev rules without inheriting udev.eclass!
 * Packages must inherit udev.eclass then call udev_reload in pkg_postinst.
```
I'll fix that, thanks!

**Pull request closed without merging**